File: JavaScript/array/MaxSubarray.js
/**
 * @param nums: A list of integers
 * @return: A integer indicate the sum of max subarray
 */
const maxSubArray = function (nums) {
  if (nums === null) {
    return 0;
  }
  
  let prefixSum = 0, sum = 0, max = Number.MIN_SAFE_INTEGER;
  for (let i = 0; i < nums.length; i++) {
    prefixSum += nums[i];
    max = Math.max(max, prefixSum - sum);
    sum = Math.min(prefixSum, sum);
  }
  return max;
}

File: Python/array/MaxSubarray.py
class Solution:
    """
    @param nums: A list of integers
    @return: A integer indicate the sum of max subarray
    """
    def maxSubArray(self, nums):
        # write your code here
        preSum = []
        temp = 0
        
        for i in range(len(nums)):
          preSum.append(temp)
          temp += nums[i]
        preSum.append(temp)  
        
        temp = preSum[0]
        maxSum = preSum[1]
        for j in range(len(preSum) - 1):
          temp = min(temp, preSum[j]) 
          maxSum = max(maxSum, preSum[j+1]-temp)
        return maxSum
          
          
  